Bottom: 511c98174d95f8ee80991d81cb6a2193db6289da
Top:    af8c233f08d37009c412a675a1478237d1c4afff
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2016-01-07 02:18:19 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 040124c..545aaac 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1136,7 +1136,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
